src/sentry/integrations/slack/message_builder/notifications.py

```diff
@@ -37,24 +39,31 @@ def build_notification_footer(notification: BaseNotification) -> str:
     return f"<{group_url}|{short_id}> via <{settings_url}|Notification Settings>"
 
 
+class SlackNotificationsMessageBuilder(SlackMessageBuilder):
+    def __init__(self, notification: BaseNotification, context: Mapping[str, Any]) -> None:
+        super().__init__()
+        self.notification = notification
+        self.context = context
+
+    def build(self) -> SlackBody:
+        if isinstance(self.notification, AlertRuleNotification):
+            return SlackIssuesMessageBuilder(
+                self.notification.group,
+                self.notification.event,
+                self.context["tags"],
+                self.notification.rules,
+                issue_alert=True,
+            ).build()
+
+        return self._build(
+            footer=build_notification_footer(self.notification),
+            text=self.context["text_description"],
+            title=self.notification.get_title(),
+        )
+
+
 def build_notification_attachment(
     notification: BaseNotification, context: Mapping[str, Any]
 ) -> Mapping[str, str]:
-    if isinstance(notification, AlertRuleNotification):
-        return build_group_attachment(
-            notification.group,
-            notification.event,
-            context["tags"],
-            notification.rules,
-            issue_alert=True,
-        )
-
-    footer = build_notification_footer(notification)
-    return {
-        "title": notification.get_title(),
-        "text": context["text_description"],
-        "mrkdwn_in": ["text"],
-        "footer_icon": get_sentry_avatar_url(),
-        "footer": footer,
-        "color": LEVEL_TO_COLOR["info"],
-    }
+    """@deprecated"""
```

Comment:
what part is deprecated, the return value?

Reply:
I refactored the message builder interface from "build_*_attachment(*args)"
to "Slack*MessageBuilder(*args).build()". Instead of updating every usage in the repo, I just use the `@deprecated` tag is to remind us not to use the old function.